engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala

```diff
@@ -293,17 +300,18 @@ final class EnsureCompiledJob(
       )
       Option.when(shouldExecute)(changeset)
     } finally {
-      ctx.locking.releasePendingEditsLock()
-      logger.log(
-        Level.FINEST,
-        s"Kept pending edits lock [EnsureCompiledJob] for ${System.currentTimeMillis() - pendingEditsLockTimestamp} milliseconds"
+      logLockRelease(
```

Comment:
Repeating this `logLockRelease` method call everywhere isn't really DRY. If the `Locking` had the `with` methods, such a logging would be centralized and more error prone.